tools/ci/gha-install.sh

```diff
@@ -35,7 +35,9 @@ fi
 # For MPI configurations, make sure mpi4py and h5py are built against the
 # correct version of MPI
 if [[ $MPI == 'y' ]]; then
-    pip install --no-binary=mpi4py mpi4py
+    # setuptools 69.4.0 causes problems with mpi4py installation
+    pip install --upgrade "setuptools<69.4.0" build wheel
```

Comment:
```suggestion
```

new setuptools are on pypi, might be worth retrying without the steuptools version limits